python/cookiecutter-django.md

```diff
@@ -0,0 +1,538 @@
+
+## Create psql database
+`$ createdb cookiecutter_starter-dev`
+
+## Load .env file for local configurations
+1. At `config/settings/base.py` set `DJANGO_READ_DOT_ENV_FILE` to load the configurations from `.env` file.
+```python
+#  config/settings/base.py
+
+#  Before
+READ_DOT_ENV_FILE = env.bool("DJANGO_READ_DOT_ENV_FILE", default=False)
+
+#  After
+READ_DOT_ENV_FILE = env.bool("DJANGO_READ_DOT_ENV_FILE", default=True)
+```
+
+2. Create `.env` file at root folder and `.env.example` to track the local configuration needed.
+```yml
+#  .env.example
+
+DJANGO_DEBUG=on
+DATABASE_URL=postgres://<user>:<password>@localhost:5432/cookiecutter_starter-dev
+```
+
+## Migrate
+`$ pipenv run python manage.py migrate`
+
+## Run test
+`$ pipenv run pytest`
+
+## Run test and get coverage percentage
+`$ pipenv run coverage run -m pytest && pipenv run coverage report -m`
+
+NOTE:
+- The command must run without exceptions because it mandatory for future configurations.
+- Don't matter if you write tests following the [unittest](https://docs.python.org/3/library/unittest.html) or [pytest](https://docs.pytest.org/) approach since `$ pipenv run pytest` command will be able to run both of them.
+
+## Install Mailhog
+`$ brew install mailhog`
+
+NOTE: If you prefer other options to install *Mailhog*, you can take a look at the [Mailhog installation guideline](https://github.com/mailhog/MailHog#installation).
+
+Start the server running `$ mailhog`
+
+Finally, if you go to `http://localhost:8025/`, your mail server is running.
+
+## Code Style
+### [flake8](https://flake8.pycqa.org/) and [flake8-isort](https://github.com/gforcada/flake8-isort)
+*NOTE: By default this package is already installed.*
+
+1. `$ pipenv install flake8 flake8-isort --dev`
+1. Add dependencies to requirement files. Below the *Code quality* separator:
+```
+flake8==<version>
+flake8-isort==<version>
+```
+3. Add custom configuration at `setup.cfg` file according to the Rootstrap's standards.
+    - **extend-ignore**: exclude flake8 warnings to avoid black conflicts. [See more.](https://black.readthedocs.io/en/stable/compatible_configs.html#id2)
+```yml
+#  setup.cfg
+
+[flake8]
+max-line-length = 120
+exclude = .tox,.git,*/migrations/*,*/static/CACHE/*,docs,node_modules,venv/*
+extend-ignore = E203, W503
+```
+4. Check running `$ pipenv run flake8`.
+    - This command list all the issues found.
+    - A useful command could be `$ pipenv run isort .` which resolves all the issues related to import styles. Also, you can add the flag `--interactive` at the end of the command to resolve or ignore one issue by one.
+
+### [black](https://black.readthedocs.io/)
+*NOTE: By default this package is already installed.*
+
+1. `$ pipenv install black --dev`
+1. Add dependencies to requirement files. Below the *Code quality* separator:
+```
+black==<version>
+```
+3. Add custom configuration at `pyproject.toml` file according to the Rootstrap's standards.
```

Comment:
Can we add here a link on Rootstrap's standards? or at least a file for example?

Reply:
I think it is being done in line 197

Reply:
Oh my bad I thought that we need more configuration lines and that was an example